File: SDKs/amazonaws.com/clouddirectory/2016-05-10/python/src/sdk/models/shared/attributekeyandvalue.py
from dataclasses import dataclass, field
from dataclasses_json import dataclass_json
from . import attributekey
from . import typedattributevalue


@dataclass_json
@dataclass
class AttributeKeyAndValue:
    key: attributekey.AttributeKey = field(default=None, metadata={'dataclasses_json': { 'field_name': 'Key' }})
    value: typedattributevalue.TypedAttributeValue = field(default=None, metadata={'dataclasses_json': { 'field_name': 'Value' }})
    

File: SDKs/amazonaws.com/clouddirectory/2016-05-10/typescript/src/sdk/models/shared/attributekeyandvalue.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { AttributeKey } from "./attributekey";
import { TypedAttributeValue } from "./typedattributevalue";



// AttributeKeyAndValue
/** 
 * The combination of an attribute key and an attribute value.
**/
export class AttributeKeyAndValue extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=Key" })
  key: AttributeKey;

  @SpeakeasyMetadata({ data: "json, name=Value" })
  value: TypedAttributeValue;
}
